schema.graphql

```diff
@@ -73,31 +79,63 @@ type VerseText {
   assessmentDetails: [AssessmentDetail]
 }
 
-enum AssessmentType {
+enum TextAnnotationType {
   SemanticSimilarity
   QAComprehensibility
   SubwordsPerSentence
+  VersePart
+  PartOfSpeech
+  NamedEntity
+}
+
+enum ReferenceAnnotationType {
+  SemanticDomain
+  KeyTerm
 }
 
 type Assessment {
   id: ID!
-  type: AssessmentType! @search(by: [exact])
+  type: TextAnnotationType! @search(by: [exact])
   bibleRevision: BibleRevision!
   reference: BibleRevision
   date: DateTime! @search
-  details: [AssessmentDetail!] @hasInverse(field: assessment)
+  details: [Annotation!] @hasInverse(field: assessment)
+}
+
+interface Annotation {
+  id: ID!
+  flag: Boolean @search
+  number: Float @search
+  tag: String @search
+  parent: [Annotation] @hasInverse(field: children)
+  children: [Annotation] @hasInverse(field: parent)
 }
 
-type AssessmentDetail {
+type TextAnnotation implements Annotation {
   id: ID!
-  assessment: Assessment!
-  assessmentMetrics: [AssessmentMetric!] @hasInverse(field: assessmentDetail)
-  verseTexts: [VerseText]
+  verseTexts: [VerseText!]
+  startPosition: Int
+  endPosition: Int
+  refVerseTexts: [VerseText]
+  refStartPosition: Int
+  refEndPosition: Int
+  flag: Boolean @search
+  number: Float @search
+  tag: String @search
+  parent: [Annotation] @hasInverse(field: children)
```

Comment:
Could you explain to me the use case for having annotations tied to other annotations?

Reply:
For example, if we have similarity at the verse level, but also have scores at the token level that are only meaningful when tied to the verse level (because they describe the contribution of that token to verse similarity, not token-to-token similarity).

Reply:
Or, if we have annotation of Parts of Speech and have a Noun Phrase, it may have multiple parts underneath (Nouns, Adjectives, Verbs)

Reply:
I think I would prefer to leave off the child/parent note attributes until we are actually integrating something like this (so we have some data tied to the need). However, I think we should document this idea/ added feature in a GH issue here.